README.md

```diff
@@ -70,6 +70,30 @@ built client from the client repository. Please consult the client's
 documentation for instructions on building the client in a development
 environment.
 
+**Tip**: If you get a **permission denied** error when running `npm link`
```

Comment:
This comes up now and then about our docs. Generally speaking I think there's a balance to be achieved between this idea of not duplicating third-party documentation, and making our own documentation user friendly. I agree that we don't want to document any third-party tool extensively when we could just link to its own docs. But I'm not averse to a couple of sentences explaining what a third-party tool is, what a command does, or the fix for a problem with a third-party tool that commonly occurs when following our instructions, etc. A lot of time I think we can make instructions unnecessarily difficult to follow by expecting people to find the third-party docs themselves, or even giving a potentially broken-in-future link to a page in the docs (and then requiring the reader to read that whole page and figure out what they need to do, even though they may not be familiar with or understand the third-party tool in question) rather than just giving the user the answer they need.

I certainly don't want our readers to have to google "npm link permission denied" to find the solution to an error message that can be reproduced by following our docs on a clean Ubuntu and that we could have supplied the solution to in our docs.

When we have a good, short third-party page to link to I think it's more grey. In this case the npm docs are more comprehensive so I'd be happy to replace the tip with a link to them.

(Although I have to say that their first suggestion of recursively changing the ownership of `/usr/local` from root sounds potentially dodgy to me, both in terms of possibly breaking multiuser systems as they point out, and also in terms of security.)